src/rulesets/Base/fastmath_able.jl

```diff
@@ -37,9 +37,16 @@ let
 
         # Unary complex functions
         @scalar_rule abs(x::Real) sign(x)
-        @scalar_rule abs2(x) 2x
+        @scalar_rule abs2(x::Real) 2x
         @scalar_rule angle(x::Real) Zero()
         @scalar_rule conj(x::Real) One()
+        function frule((Δx,), abs, x::Complex)
```

Comment:
@sethaxen  shouldn't this be

```julia
frule((_, Δx), abs, x::Complex)
```
?

Reply:
"needs tests"?

Reply:
@MasonProtter Of course, yes, you're right. 🤦